### DISCUSSION TITLE
How to use a Demux/Set Pids for DVB/SAT>IP Streams (TS) ?

### DISCUSSION THREAD

**Discussion by tantegrete** [FactoryXCode/MfPack#60] [General]
I'm using DirectShow for many years now and it still works very well receiving DVB/SAT>IP streams.
Beeing also very experiened in bulding several types of graphs for nearly every usage in DS.
How to do that similar in MF ? ..... Haven't found any "Tutorial for dummies"  yet....
Is it possible using well working codecs like LAV-Filters, renderes and my own push-sources ?
Please help and give some hints / links.
Using Delphi 11.3 CE.

**Comment by FactoryXCode** [comment]
Hi,
The last time I build such, it was on DirectShow and it still works on Windows 11. So, why change?
DirectShow will not be deprecated in a short time. Developers are however encouraged to switch to Media Foundation.
Now, being said that, before using Media Foundation development, you should install the latest Windows SDK, to be able to use some tools, like TopoEdit which is almost similar to GraphEdit for DirectShow.  TopoEdit will give you a good startingpoint how to implement the necessary interfaces.
Did you check the Microsoft examples how to deal with tv and / or satellite tuner channels?
On Media Foundation a DMO is called a MFT, a lav filter is a transform (see transforms or [MFT's](https://learn.microsoft.com/en-us/windows/win32/medfound/media-foundation-transforms)  an example about how to use MFT's you may find [here](https://github.com/FactoryXCode/MfPack/tree/Master/MfPack/Samples/MfTranscode). I suggest you should read about MFT's on [MS  Learn](https://learn.microsoft.com/en-us/windows/win32/medfound/media-foundation-transforms).
However, mixing DirectShow and Media Foundation is not recommended and most of the time is not possible. I think you have to write custom filters (DMO's and MFT's) using the Media Foundation API. 
Regards, Tony.
